SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp

```diff
@@ -247,7 +248,22 @@ void USpatialReceiver::OnAddComponent(const Worker_AddComponentOp& Op)
 		// This means we need to be inside a critical section, otherwise we may not have all the requisite
 		// information at the point of creating the Actor.
 		check(bInCriticalSection);
-		PendingAddActors.AddUnique(Op.entity_id);
+
+		// PendingAddActor should only be populated with actors we actually want to check out.
+		// So a local and ready actor should not be considered as an actor pending addition.
+		// Nitty-gritty implementation side effect is also that we do not want to stomp
+		// the local state of a not ready-actor, because it is locally authoritative and will remain
+		// that way until it is marked as ready. Putting it in PendingAddActor has the side effect
+		// that the received component data will get dropped (likely outdated data), and is
+		// something we do not wish to happen for ready actor (likely new data received through
+		// a component refresh on authority delegation).
+		{
+			AActor* EntityActor = Cast<AActor>(PackageMap->GetObjectFromEntityId(Op.entity_id));
+			if (EntityActor == nullptr || !EntityActor->IsActorReady())
```

Comment:
I have been trying to think of a case where this could break... but I think we are good for now.
Just in case you can find something: the case I was thinking about was a bNetLoadOnClient actor that would be set ready and then received again at some other point. But I think we are good in all the cases I can think of.

Reply:
I think ready should by definition mean that this is true and if it doesn't we should probably treat that as a bug with readiness. Although net load on client and initial dormancy both have a tendency of throwing up weird cases.